src/transformers/generation_utils.py

```diff
@@ -1214,13 +1710,33 @@ def beam_sample(
         while cur_len < max_length:
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
-            outputs = self(**model_inputs, return_dict=True)
+            outputs = self(
+                **model_inputs,
+                return_dict=True,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
             next_token_logits = outputs.logits[:, -1, :]
 
             # adjust token scores (a no-op by default)
             next_token_logits = self.adjust_logits_during_generation(
                 next_token_logits, cur_len=cur_len, max_length=max_length
             )
+            # Store logits, attentions and hidden_states when required
+            if return_dict_in_generate:
+                if output_scores:
+                    logits += (next_token_logits,)
```

Comment:
Here we should prob add `next_token_scores` after 
`next_token_scores = next_token_scores.view(batch_size, num_beams * vocab_size)`